File: Stowaway.Backend/Market.Shared/Constants/Permissions.cs
namespace Market.Shared.Constants;

public static class Permissions
{
    // Non permissions
    public const string ClaimType = "permission";
    public const string AuthPrefix = "Permission:";

    // User permissions -> Admin permissions
    public const string UsersRead = "Users.Read";
    public const string UsersCreate = "Users.Create";
    public const string UsersUpdate = "Users.Update";
    public const string UsersDelete = "Users.Delete";

    //User self permissions -> When a user wants to get their own data, update own fields, delete own account
    public const string UsersSelfRead = "Users.Self.Read";
    public const string UsersSelfUpdate = "Users.Self.Update";
    public const string UsersSelfDelete = "Users.Self.Delete";

    //Role Permissions
    public const string RolesRead = "Roles.Read";
    public const string RolesCreate = "Roles.Create";
    public const string RolesUpdate = "Roles.Update";
    public const string RolesDelete = "Roles.Delete";
    //Warehouse Permissions
    public const string WarehouseCreate = "Warehouse.Create";
    public const string WarehouseRead = "Warehouse.Read";
    public const string WarehouseUpdate = "Warehouse.Update";
    public const string WarehouseDelete = "Warehouse.Delete";

    //Container Permissions
    public const string ContainerRead = "Container.Read";
    public const string ContainerCreate = "Container.Create";
    public const string ContainerUpdate = "Container.Update";
    public const string ContainerDelete = "Container.Delete";

    //Item Permissions
    public const string ItemRead = "Item.Read";
    public const string ItemCreate = "Item.Create";
    public const string ItemUpdate = "Item.Update";
    public const string ItemDelete = "Item.Delete";

    //Tag Permissions
    public const string TagRead = "Tag.Read";
    public const string TagCreate = "Tag.Create";

    //Order Permissions
    public const string OrderRead = "Order.Read";
    public const string OrderCreate = "Order.Create";
    public const string OrderUpdate = "Order.Update";
    public const string OrderDelete = "Order.Delete";

    //Supplier Permissions
    public const string SupplierRead = "Supplier.Read";
    public const string SupplierCreate = "Supplier.Create";
    public const string SupplierUpdate = "Supplier.Update";
    public const string SupplierDelete = "Supplier.Delete";

    //Cart Permissions (self-scoped)
    public const string CartManage = "Cart.Manage";

    //Warehouse users / priviledge group management
    public const string WarehouseUsersManage = "WarehouseUsers.Manage";
}

File: Stowaway.Frontend/src/app/shared/constants/permissions.ts
// Mirrors Stowaway.Backend/Stowaway.Shared/Constants/Permissions.cs — keep in sync.
export const Permissions = {
  UsersRead: 'Users.Read',
  UsersCreate: 'Users.Create',
  UsersUpdate: 'Users.Update',
  UsersDelete: 'Users.Delete',

  UsersSelfRead: 'Users.Self.Read',
  UsersSelfUpdate: 'Users.Self.Update',
  UsersSelfDelete: 'Users.Self.Delete',

  RolesRead: 'Roles.Read',
  RolesCreate: 'Roles.Create',
  RolesUpdate: 'Roles.Update',
  RolesDelete: 'Roles.Delete',

  WarehouseRead: 'Warehouse.Read',
  WarehouseCreate: 'Warehouse.Create',
  WarehouseUpdate: 'Warehouse.Update',
  WarehouseDelete: 'Warehouse.Delete',

  ContainerRead: 'Container.Read',
  ContainerCreate: 'Container.Create',
  ContainerUpdate: 'Container.Update',
  ContainerDelete: 'Container.Delete',

  ItemRead: 'Item.Read',
  ItemCreate: 'Item.Create',
  ItemUpdate: 'Item.Update',
  ItemDelete: 'Item.Delete',

  TagRead: 'Tag.Read',
  TagCreate: 'Tag.Create',

  OrderRead: 'Order.Read',
  OrderCreate: 'Order.Create',
  OrderUpdate: 'Order.Update',
  OrderDelete: 'Order.Delete',

  SupplierRead: 'Supplier.Read',
  SupplierCreate: 'Supplier.Create',
  SupplierUpdate: 'Supplier.Update',
  SupplierDelete: 'Supplier.Delete',

  CartManage: 'Cart.Manage',

  WarehouseUsersManage: 'WarehouseUsers.Manage',
} as const;
